### PR TITLE
REL, MAINT: prepare for 1.10.1

### PR DESCRIPTION
**Pull request by tylerjereddy**
* the usual preparation for the next point release on the maintenance branch

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -4,7 +4,7 @@ project(
   # Note that the git commit hash cannot be added dynamically here (it is added
   # in the dynamically generated and installed `scipy/version.py` though - see
   # tools/version_utils.py
-  version: '1.10.0',
+  version: '1.10.1',
```
Comment (rgommers) [inline meson.build:7]: `'1.10.1.dev0'` here

**Review by rgommers — APPROVED** [review]
LGTM, one minor comment about version string

File: meson.build (hunk 1 above)
Comment (tylerjereddy) [reply meson.build:7]: has this changed from previous approach?

- https://github.com/scipy/scipy/pull/17263
- https://github.com/scipy/scipy/pull/17176

Comment (rgommers) [reply meson.build:7]: No, I just missed it in reviewing there. The `setup.py` equivalent uses `ISRELEASED = False` which also adds `.dev0` to the version string. And in `main` it has the same `.dev0` in `meson.build`.

**Comment by tylerjereddy** [comment]
I addressed the minor version number comment and CI is all green so far--I don't think we need to wait for the final 3 Azure jobs for the version number adjustments.

**Pull request merged**